packages/components/src/TreeLikeList/CheckedTreeItem/CheckedTreeItem.tsx

```diff
@@ -0,0 +1,86 @@
+import { Collapse } from '../../Collapse';
+import { Checkbox } from '../../Checkbox';
+import { FormControlLabel } from '../../FormControlLabel';
+import { type CheckedTreeItemProps } from '../types';
+import { ChevronIcon, CollapseButton, InnerList, Item } from '../styles';
+
+import { useLogic } from './useLogic';
+import { ItemContent } from './styles';
+
+export const CheckedTreeItem = ({
+  id,
+  label,
+  level,
+  componentList = 'ul',
+  componentItem = 'li',
+  listProps,
+  itemProps,
+  children = [],
+  value = [],
+  onChange,
+}: CheckedTreeItemProps) => {
+  const { isOpen, isSelected, isIndeterminate, handleToggle, handleChange } =
+    useLogic({ id, value, children, onChange });
+
+  if (children.length) {
+    return (
+      <Item {...itemProps} $level={level} as={componentItem}>
+        <ItemContent
+          $isSelected={isSelected}
+          $level={level}
+          onClick={handleChange}
+        >
+          <CollapseButton variant="text" onClick={handleToggle}>
+            <ChevronIcon $isActive={isOpen} />
+          </CollapseButton>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={isSelected}
+                indeterminate={isSelected ? false : isIndeterminate}
+              />
+            }
+            label={label}
+            onChange={handleChange}
+            onClick={(event) => event.stopPropagation()}
```

Comment:
https://kaluga-astral.github.io/style-guide/docs/rules/react/logic#%D0%B7%D0%B0%D0%BF%D1%80%D0%B5%D1%89%D0%B5%D0%BD%D0%B0-%D1%80%D0%B5%D0%B0%D0%BB%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F-%D0%BE%D0%B1%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D1%87%D0%B8%D0%BA%D0%BE%D0%B2-%D0%B2-jsx